src/partials/messenger.ts

```diff
@@ -942,7 +942,24 @@ class NavigationController {
     }
 
     public contacts(): threema.ContactReceiver[] {
-        return Array.from(this.webClientService.contacts.values()) as threema.ContactReceiver[];
+        const contacts = [];
+        // Since `values()` on a map returns an iterator, we cannot directly
+        // apply the `.filter()` method that arrays provide. Therefore, in
+        // order to avoid creating an intermediate array just for filtering,
+        // create the list of contacts imperatively.
+        for (const contact of this.webClientService.contacts.values()) {
```

Comment:
Why not .filter()?

Reply:
You mean on the map directly? Would be possible, but on the other hand we never need the keys for filtering (and `values()` ignores those completely).

If you meant `.values().filter()`, then see the comment above :slightly_smiling_face:

Reply:
Oh, I completely missed that. We *could* use generators and `yield` but since we only want an array as a result this would probably be overkill.

Reply:
TIL that generators exist in JS... https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function* But yes, there's no advantage since we'll convert to an array anyways.